java/src/main/java/com/google/cloud/dataproc/templates/jdbc/JDBCToJDBC.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (C) 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.dataproc.templates.jdbc;
+
+import com.google.cloud.dataproc.templates.BaseTemplate;
+import com.google.cloud.dataproc.templates.util.PropertyUtil;
+import com.google.cloud.dataproc.templates.util.ValidationUtil;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.HashMap;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.SparkSession;
+import org.apache.spark.sql.execution.datasources.jdbc.JDBCOptions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class JDBCToJDBC implements BaseTemplate {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(JDBCToJDBC.class);
+  private final JDBCToJDBCConfig config;
+
+  public JDBCToJDBC(JDBCToJDBCConfig config) {
+    this.config = config;
+  }
+
+  public static JDBCToJDBC of(String... args) {
+    JDBCToJDBCConfig config = JDBCToJDBCConfig.fromProperties(PropertyUtil.getProperties());
+    JDBCToJDBC configObject = new JDBCToJDBC(config);
+    configObject.validateInput();
+    LOGGER.info("Config loaded\n{}", config);
+    return configObject;
+  }
+
+  @Override
+  public void validateInput() {
+    ValidationUtil.validateOrThrow(config);
+  }
+
+  @Override
+  public void runTemplate() {
+    try (SparkSession spark = SparkSession.builder().appName("JDBC to JDBC").getOrCreate()) {
+      // Set log level
+      spark.sparkContext().setLogLevel(config.getSparkLogLevel());
+      Dataset<Row> inputData = load(spark);
+      write(inputData);
+
+      if (StringUtils.isNotBlank(config.getJdbcOutputPrimaryKey())) {
+        addPrimaryKeyColumn();
+      }
```

Comment:
Primary key should only be added if no user transformation has been performed.
`inputData = spark.sql(config.getJdbcTempSQLQuery());`